ui/tests/acceptance/clients/counts/overview-test.js

```diff
@@ -153,79 +148,98 @@ module('Acceptance | clients | overview', function (hooks) {
       );
   });
 
-  test('totals filter correctly with full data', async function (assert) {
+  test('shows running totals with monthly breakdown charts', async function (assert) {
     assert
       .dom(CHARTS.container('Vault client counts'))
       .exists('Shows running totals with monthly breakdown charts');
     assert.dom(CLIENT_COUNT.attributionBlock).exists('Shows attribution area');
+  });
 
-    const response = await this.store.peekRecord('clients/activity', 'some-activity-id');
-    // FILTER BY NAMESPACE
-    await clickTrigger();
-    await searchSelect.options.objectAt(0).click();
-    await settled();
-    const topNamespace = response.byNamespace[0];
-    const topMount = topNamespace.mounts[0];
-
-    assert.dom(CLIENT_COUNT.selectedNs).hasText(topNamespace.label, 'selects top namespace');
-    assert.dom('[data-test-top-attribution]').includesText('Top auth method');
-    assert
-      .dom('[data-test-attribution-clients] p')
-      .includesText(`${formatNumber([topMount.clients])}`, 'top attribution clients accurate');
-
-    let expectedStats = {
-      Entity: formatNumber([topNamespace.entity_clients]),
-      'Non-entity': formatNumber([topNamespace.non_entity_clients]),
-      ACME: formatNumber([topNamespace.acme_clients]),
-      'Secret sync': formatNumber([topNamespace.secret_syncs]),
-    };
-    for (const label in expectedStats) {
+  module('filtering charts', function () {
```

Comment:
Was the decision to convert to modules to try and address the flakiness? In my experience, these tests weren't flaky before 🤔 , so if that isn't the case then could this refactor happen in a separate PR? 

I completely agree with refactoring these into modules! But a separate PR would be make this one easier to review and check that test coverage has been added for the relevant logic updates. Historically, we've also had folks accidentally remove important test assertions for this area of the code base due to the intricacies of this dang dashboard 😞

Reply:
that's correct, i moved them into separate tests to try to isolate the failures. what happened here was that i switched to using the `sync` mirage handler instead of the `clients` one at the top-level `beforeEach` hooks because making assertions on the secrets sync charts required secrets sync being included in the `/activated-features` response, which the `clients` handler didn't have. similarly, i assumed the `sync` handler would also be better to use here since it would include sync client data to populate the charts. however when i made this switch, the tests failed inconsistently, so i split the tests up to try and figure out if it was only specific aspects of the tests or if it was the whole ding-dang thing. 😂

as it turns out, the root of the problem lies somewhere in the differences between the `sync` and `clients` handlers, because yeah, splitting up the module didn't help much at all. i am reverting this change today and am investigating other options (potentially leaving the sync assertions out of these tests and instead making those assertions in a completely separate module so we can use the `sync` handler instead of `clients`). 🤔

Reply:
i looked at this for quite awhile today and these chart / chart filtering tests seem to be flaky; at least on our sidebranch. check out the video i left in slack for more (github wouldn't let me upload it here because the file size was too big 🙁 )

@Monkeychip and i talked about it while pairing today and assuming these tests are failing on other branches, we'll need to tackle the fixes in a separate PR.